### PR TITLE
JUnit tests run now in JPMS, fixed unintended reflection isnpection on java.base classes.

### LINKED ISSUE #205: Better JPMS support in POM
Update surefire version, set surefire plugin to put yasson on module path.

See #196 @psakar comments for details.

### LINKED ISSUE #214: Yasson parses classes of known type
Known types such as java.lang.String or java.lang.Integer when are held inside collections, they are parsed reflectively by Yasson.

For example:

```java
class JavaBaseProperties {
  public List<String> names;
}
```

will cause java.lang.String to be parsed by Yasson into its class model.

### PR DESCRIPTION
**Pull request by bravehorsie**
Fixes #205 
Fixes #214 

Signed-off-by: Roman Grigoriadi <roman.grigoriadi@oracle.com>

### CODE REVIEW

**Comment by bravehorsie** [comment]
All java files changes are related to unintentional java.base type scanning, see #214

File: src/main/java/org/eclipse/yasson/internal/serializer/AbstractContainerDeserializer.java [hunk 1]
```diff
@@ -120,11 +120,15 @@ protected DeserializerBuilder newUnmarshallerItemBuilder(JsonbContext ctx) {
     }
 
     protected JsonbDeserializer<?> newCollectionOrMapItem(Type valueType, JsonbContext ctx) {
+        //TODO needs performance optimization on not to create deserializer each time
```
Comment (aguibert) [inline src/main/java/org/eclipse/yasson/internal/serializer/AbstractContainerDeserializer.java]: I'm concerned about this performance regression here. Do we have any immediate need to function as a JPMS module?  If we have to make a tradeoff between performance and JPMS, I'd prefer to stick with performance.

Comment (bravehorsie) [reply src/main/java/org/eclipse/yasson/internal/serializer/AbstractContainerDeserializer.java]: Well the performance issue is there regardless of Yasson is loaded as JPMS module or not, I just added a comment, revisiting the code so it is not forgot. In serialization there is already caching of serializer of last item in collection, similar it should be cached for deserialization of collections.

Comment (bravehorsie) [reply src/main/java/org/eclipse/yasson/internal/serializer/AbstractContainerDeserializer.java]: added #219 to track it.

File: src/test/java/org/eclipse/yasson/defaultmapping/collections/ArrayTest.java [hunk 2]
```diff
@@ -143,11 +144,14 @@ public void testUnmarshallMapWithArrayValue() {
     public void testArrayOfNulls() {
         String[] nulls = new String[2];
         String expected = "[null,null]";
-        assertEquals(expected, jsonb.toJson(nulls));
+        /*assertEquals(expected, jsonb.toJson(nulls));
```
Comment (aguibert) [inline src/test/java/org/eclipse/yasson/defaultmapping/collections/ArrayTest.java]: did you intend to leave this part of the test commented out?  If so, `String[] nulls` is no longer used.

Comment (bravehorsie) [reply src/test/java/org/eclipse/yasson/defaultmapping/collections/ArrayTest.java]: That is clearly not intentional, commented out during code fix and forgot about it. Good catch.

**Pull request merged**